### PR TITLE
DSD-946: Helper text position

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket [DSD-946.](https://jira.nypl.org/browse/DSD-946)

## This PR does the following:

- The PR for [DSD-824](https://github.com/NYPL/nypl-design-system/pull/925) necessitated updates to the `helperErrorText` theme to correctly display when used with a `Select` element with a `labelPosition` of inline.

## How has this been tested?

- Locally, through Storybook.

## Accessibility concerns or updates

- For this update, I added two props for internal use to `HelperErrorText` – `isInlined` and `labelWidth`. I removed them from the Storybook props table. Edwin, Marty, you can let me know if there's another way you'd like me to accomplish this update.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr974-yhtib5mecteyjojgom92uawruxwiuwx4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266eb748a4c5796e7da6929

**Review by EdwinGuzman — COMMENTED** [review]
I'd prefer if you pass the style prop to `HelperErrorText` than to add two additional props that are limited in use.

I think this should work
```
<HelperErrorText
              id={`${id}-helperText`}
              isInvalid={isInvalid}
              text={footnote}
              ml={["auto", "auto", `${labelWidth}px`]}
            />
```

This _might_ not be a problem but if you switch between inline and default, the helper text does not update:
Screen Shot 2022-04-25 at 3 01 30 PM

I think it's because `labelWidth` doesn't get reset when you go from `LabelPositions.Inline` to `LabelPositions.Default`. Add that check in the `useEffect` in the `Select` component so it reverts back to the correct initial value.

**Review by bigfishdesign13 — COMMENTED** [review]
The problem that @EdwinGuzman noted only happens on the `Canvas` tab.  The positioning never gets stuck under the `Docs` tab.  Why would that be?

Other than that, it looks good.  Make the one change that Edwin asked for and then tyhis should be good to go.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr974-yhtib5mecteyjojgom92uawruxwiuwx4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266eb748a4c5796e7da6929

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -107,6 +107,10 @@ export const Select = chakra(
       // must be passed.
       const controlledProps = onChange ? { onChange, value } : {};
 
+      // The number of pixels between the label and select elements
+      // when the labelPosition is inline (equivalent to --nypl-space-xs).
+      const labelSelectGap = 8;
```
Comment (EdwinGuzman) [inline src/components/Select/Select.tsx:112]: 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -63,6 +63,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 - Updates storybook URL from /storybook-static/ to /reservoir/v0_26.
 - Updates all references to "design system" in the Storybook documentation to include "Reservoir" in the name of the design system.
 - Updates language on the `Welcome` page.
+- Updates the `HelperErrorText` styling to correctly display when used with a `Select` element with a `labelPosition` of inline.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: This should go under the "## Prerelease" section.

File: src/components/Select/Select.stories.mdx [hunk 3]
```diff
@@ -115,7 +115,7 @@ is left blank, a value will be generated for you.
       isDisabled: false,
       isInvalid: false,
       isRequired: false,
-      labelPosition: "LabelPositions.Default",
+      labelPosition: "LabelPositions.Inline",
```
Comment (EdwinGuzman) [inline src/components/Select/Select.stories.mdx]: Set the initial value to `Default`. I think that's the most commonly used layout rather than inline.

**Review by EdwinGuzman — COMMENTED** [review]
Looks good just minor updates.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr974-yhtib5mecteyjojgom92uawruxwiuwx4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266eb748a4c5796e7da6929

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 ### Updates
 
 - Updates storybook URL from /reservoir/v0_26 to /reservoir/v0.
+- Updates the `HelperErrorText` styling to correctly display when used with a `Select` element with a `labelPosition` of inline.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: This needs to be moved up under the "## Prerelease" section. This is under the already released 0.26.1 version. When you merge in the `development` branch, you'll see that there are now new additions in the CHANGELOG and you can add this to the proper area.

**Review by EdwinGuzman — COMMENTED** [review]
The changelog should be updated. Should this go into the next `0.27.0` release or wait until `0.28.0`, @bigfishdesign13 ?

Once we decide on the version, update the version for the `Select` component in its Storybook version table.

**Comment by bigfishdesign13** [comment]
> The changelog should be updated. Should this go into the next `0.27.0` release or wait until `0.28.0`, @bigfishdesign13 ?
> 
> Once we decide on the version, update the version for the `Select` component in its Storybook version table.

@EdwinGuzman and @oliviawongnyc,
If this update is ready to go - outside of the minor CHANGELOG and Latest Version number changes - then lets get it into `0.27.0`.  Those final changes should be quick.

**Review by EdwinGuzman — APPROVED** [review]
Looks good. I made updates to `development`'s `package-lock.json` file so once this is updated and tugboat passes, this can be merged in.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr974-yhtib5mecteyjojgom92uawruxwiuwx4.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266eb748a4c5796e7da6929

**Pull request merged**